src/target/ext/register_info.rs

```diff
@@ -0,0 +1,220 @@
+//! Provide register information for the target. See the lldb-gdb-remote [documentation](https://github.com/llvm/llvm-project/blob/main/lldb/docs/lldb-gdb-remote.txt) for more information.
+//!
+//!_Note:_ LLDB specific! This packet is meant as an alternative to the
+//! 'qXfer:features:read:target.xml' packet. Some targets don't have register
+//! context in the compiled version of the debugger. Help the debugger by
+//! dynamically supplying the register info from the target. The debugger will
+//! request the register info in a sequential manner till an error packet is
+//! received. Register info search order:
+//!     1 - Use the target definition python file if one is specified.
+//!     2 - If the target definition doesn't have any of the info from the
+//!     target.xml (registers) then proceed to read the target.xml.
+//!     3 - Fall back on the qRegisterInfo packets.
+//!     4 - Use hardcoded defaults if available.
+//! (Taken from the LLDB [source](https://github.com/llvm/llvm-project/blob/369ce54bb302f209239b8ebc77ad824add9df089/lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp#L397-L402)
+//!
+//! sSee issue [#99](https://github.com/daniel5151/gdbstub/issues/99) for more info on LLDB compatibility.
+use crate::target::Target;
+
+/// Describes the register info for a single register of the target
+pub struct Register<'a> {
+    /// The primary register name
+    pub name: &'a str,
+    /// An alternate name for the register
+    pub alt_name: Option<&'a str>,
+    /// Size in bits of a register (base 10)
+    pub bitsize: usize,
+    /// The offset within the 'g' and 'G' packet of the register data for this
+    /// register (base 10)
+    pub offset: usize,
+    /// The encoding type of the register
+    pub encoding: Encoding,
+    /// The preferred format for display of this register
+    pub format: Format,
+    /// The register set name this register belongs to
+    pub set: &'a str,
+    /// The GCC compiler registers number for this register
+    pub gcc: Option<usize>,
+    /// The DWARF register number for this register that is used for this
+    /// register in the debug information
+    pub dwarf: Option<usize>,
+    /// Specify as a generic register
+    pub generic: Option<Generic>,
+    /// Other concrete register values this register is contained in
+    pub container_regs: Option<&'a [usize]>,
+    /// Specifies which register values should be invalidated when this register
+    /// is modified
+    pub invalidate_regs: Option<&'a [usize]>,
+}
+
+/// Describes the encoding type of the register
+#[non_exhaustive]
+pub enum Encoding {
+    /// unsigned integer
+    Uint,
+    /// signed integer
+    Sint,
+    /// IEEE 754 float
+    IEEE754,
+    /// vector register
+    Vector,
+}
+
+impl Encoding {
+    /// Match an enum field to its corresponding string slice
+    pub fn as_str(&self) -> &str {
```

Comment:
Unless I'm missing something, I don't see why an end user would ever want to obtain the specific backing string corresponding to an enum variant. I suspect these functions should just be inline `match` statements within the core_impl code, and not part of the public API